crates/ruma-common/tests/events/relations.rs

```diff
@@ -0,0 +1,217 @@
+    );
+}
+
+#[test]
+fn reply_serialize() {
+    let content = assign!(RoomMessageEventContent::text_plain("This is a reply"), {
+        relates_to: Some(Relation::Reply { in_reply_to: InReplyTo::new(event_id!("$1598361704261elfgc").to_owned()) }),
+    });
+
+    assert_eq!(
+        to_json_value(content).unwrap(),
+        json!({
+            "msgtype": "m.text",
+            "body": "This is a reply",
+            "m.relates_to": {
+                "m.in_reply_to": {
+                    "event_id": "$1598361704261elfgc",
+                },
+            },
+        })
+    );
+}
+
+#[test]
+#[cfg(feature = "unstable-msc2676")]
+fn replacement_serialize() {
+    use ruma_common::events::room::message::Replacement;
+
+    let content = assign!(
+        RoomMessageEventContent::text_plain("<text msg>"),
+        {
+            relates_to: Some(Relation::Replacement(
+                Replacement::new(
+                    event_id!("$1598361704261elfgc").to_owned(),
+                    Box::new(RoomMessageEventContent::text_plain("This is the new content.")),
+                )
+            ))
+        }
+    );
+
+    assert_eq!(
+        to_json_value(content).unwrap(),
+        json!({
+            "msgtype": "m.text",
+            "body": "<text msg>",
+            "m.new_content": {
+                "body": "This is the new content.",
+                "msgtype": "m.text"
+            },
+            "m.relates_to": {
+                "rel_type": "m.replace",
+                "event_id": "$1598361704261elfgc",
+            },
+        })
+    );
+}
+
+#[test]
+#[cfg(feature = "unstable-msc2676")]
+fn replacement_deserialize() {
+    use ruma_common::events::room::message::Replacement;
+
+    let json = json!({
+        "msgtype": "m.text",
+        "body": "<text msg>",
+        "m.new_content": {
+            "body": "Hello! My name is bar",
+            "msgtype": "m.text"
+        },
+        "m.relates_to": {
+            "rel_type": "m.replace",
+            "event_id": "$1598361704261elfgc",
+        },
+    });
+
+    assert_matches!(
+        from_json_value::<RoomMessageEventContent>(json).unwrap(),
+        RoomMessageEventContent {
+            msgtype: MessageType::Text(_),
+            relates_to: Some(Relation::Replacement(Replacement { event_id, new_content, .. })),
+            ..
+        } if event_id == "$1598361704261elfgc"
+          && matches!(&new_content.msgtype, MessageType::Text(text) if text.body == "Hello! My name is bar")
+    );
+}
+
+#[test]
+#[cfg(feature = "unstable-msc3440")]
+fn thread_plain_serialize() {
+    use ruma_common::events::room::message::Thread;
+
+    let content = assign!(
+        RoomMessageEventContent::text_plain("<text msg>"),
+        {
+            relates_to: Some(Relation::Thread(
+                Thread::plain(
+                    event_id!("$1598361704261elfgc").to_owned(),
+                    event_id!("$latesteventid").to_owned(),
+                )
+            ))
```

Comment:
Just to keep with the convention of having trailing commas where further list items may be added on subsequent lines:

```suggestion
                ),
            )),
```

Reply:
Same below in a few places.